diesel_migrations/migrations_macros/src/embed_migrations.rs

```diff
@@ -99,8 +108,70 @@ fn migration_literal_from_path(path: &Path) -> Result<proc_macro2::TokenStream,
     let sql_file = path.join("up.sql");
     let sql_file_path = sql_file.to_str();
 
+    let metadata_file = path.join("metadata.toml");
+    let metadata = if metadata_file.exists() {
+        let metadata = migration_metadata_from_path(&metadata_file)?;
+        quote!(Some(#metadata))
+    } else {
+        quote!(None)
+    };
+
     Ok(quote!(&EmbeddedMigration {
         version: #version,
         up_sql: include_str!(#sql_file_path),
+        metadata: #metadata
     }))
 }
+
+fn migration_metadata_from_path(path: &Path) -> Result<proc_macro2::TokenStream, Box<dyn Error>> {
+    let mut buf = Vec::new();
+    let mut file = File::open(path)?;
+    file.read_to_end(&mut buf)?;
+    let value = TomlMetadata::from_slice(&buf)?.0;
```

Comment:
I think it is clearer and much shorter to just do the following:
```rust
quote! {
    TomlMetadata::from_slice(#buf)?
}
```
especially because we fallback to string paring  for datetime.